shared/workspace_repo/api_client.go

```diff
@@ -172,7 +178,7 @@ func (c *WorkspaceRepo) enforceClient(ctx context.Context) (*tfe.Client, error)
 					return nil, errors.Wrap(err, "provided tfe token not valid, and cannot refresh it in a non-interactive mode")
 				}
 				errs = multierror.Append(errs, err)
-				state = tokenRefreshNeeded
+				state = tokenMissing
```

Comment:
Token refresh only refreshes an existing token by logging user into TFE; if the token was previously invalidated, it will just keep prompting to open the browser window. Setting the state to tokenMissing if token still doesn't work will force tfeLogin().